src/languages/asciidoc.js

```diff
@@ -107,17 +107,38 @@ export default function(hljs) {
         begin: '^(NOTE|TIP|IMPORTANT|WARNING|CAUTION):\\s+',
         relevance: 10
       },
-      // inline strong
+      // inline unconstrained strong (single line)
       {
         className: 'strong',
-        // must not follow a word character or be followed by an asterisk or space
-        begin: '\\B\\*(?![\\*\\s])',
-        end: '(\\n{2}|\\*)',
-        // allow escaped asterisk followed by word char
-        contains: [{
-          begin: '\\\\*\\w',
-          relevance: 0
-        }]
+        begin: /\*\*(.+?)\*\*/
+      },
+      // inline unconstrained strong (multi-line)
+      {
+        className: 'strong',
+        begin: /\*{2}(\s\S+)*\*{2}/,
+        relevance: 0
+      },
+      // escaped formatting marks (i.e., \* \_ or \`)
+      {
+        begin: /\\[*_`]/
+      },
+      // inline contrained string (single line)
+      {
+        className: 'strong',
+        // must not precede or follow a word character
+        begin: /\B\*\w+\*\B/,
```

Comment:
Too tight.

```asciidoc
I know they say that *"bob loves jemma"*.
```

Why not `[^\n]*`?

Reply:
I'm not sure these single expressions aren't going to cut it... we'll need separate begin and end tags to handle things like:

```asciidoc
Does *Bob _love_ Gemma*?
```
Which we should handle (we do for Markdown).    Or even the proper handling of `\*` inside bold, which you removed.  Generally we try not to merge a PR that removes features unless there are no other options.

Reply:
In generally anytime you try and match the middle it's hard.  In case like this you'd want to be super flexible, like I suggest earlier... for example if we're matching a single line bold:

```
// doesn't match all sorts of things, `"` just to name one.
 begin: /\B\*\w+\*\B/,
// matches everything except a new line, the one thing we know isn't allowed
 begin: /\B\*[^\n]+?\*\B/,
```

But again you'd have to split this into begin end still:

```js
begin: /\B\*(?=[^\n]+?\*\B)/, // look-ahead to make sure it closes
contains: OTHER_STUFF,
end: /\*\B/
```

Of course this gets harder since the regex also has to handle escapes, like `\*`...

Reply:
> Of course this gets harder since the lookahead regex also has to correctly handle escapes, like \*...

Yes that's why I thought it would be simpler to match everything in `begin`... :unamused: 

> we'll need separate begin and end tags to handle things like (emphasis inside bold)

Unless I missed something, it currently does not work.
It would be nice to support it but we should probably to do it in another pull request.

Reply:
> Or even the proper handling of \* inside bold, which we supported previously.

I removed it because I don't see why we need it:

```
This stuff sure is *gr*e*e*dy*!
```

This stuff sure is <strong>\*gr\*e\*e\*dy\*</strong>!

> I'm not convinced yet we can solve the * problem 100%,

I don't think it will be possible unless we create a parser, that's why I'm pruning uncommon syntax and/or syntax with undefined behavior.

I think it's important that `This is not \*strong*!` works.
Please note that `This is *str\*ong*!` will produce: This is <strong>\*str\\\*ong*</strong>!

Reply:
We should probably port tests from Asciidoctor: https://github.com/asciidoctor/asciidoctor/blob/master/test/substitutions_test.rb

```
\**Git**Hub
```

Result: <strong>*Git</strong>*Hub

```
\*a few strong words*
```

Result:  \*a few strong words\*

As you can see `\*` inside bold is not tested.

Reply:
> Or even the proper handling of * inside bold, which we supported previously.

I was referring to `\*`, sorry for not being clearer.  Someone went to the trouble of supporting it before making me reason perhaps it's more important than you think. :-)  Obviously * inside bold doesn't require special treatment if it isn't triggering the end of the bold rule or causing any issues.

> Yes that's why I thought it would be simpler to match everything in begin... 

Which is the right idea sort of but you have to cover the whole expression and do it with a look-ahead... so that the actual MATCH portion of begin and end is "*" and "*".  But that's not the largest issue - see my points below on regressions.

> I don't think it will be possible unless we create a parser

The regex I gave you was working for simple cases, did you try it or play around with it in regex101?  See the attached snap...

> It would be nice to support it but we should probably to do it in another pull request.

Oh I'm not trying to say accomplish everything in this PR (far from it!), but I'm also not wanting to merge anything that I know is still broken or is going to make future work harder. So regarding broken lets go back to my prior example:

```
Does *Bob _love_ Gemma*?
```

This probably sort of works thanks to _ being in \w... So my apologies for not using a better example:

```
Does *Bob -love- Gemma*? (broken -)
Does *Bob "love" Gemma*? (broken ")
Does *Bob love Gemma? Truly?* (broken ?)
```

Your PR would break all these that worked previously, would it not?  It's possible that I could be persuaded to drop `\*` for the short-term, but even if that were true I still think there is work to be done here before this would be mergeable.  Go back and look at my comments regarding `[^\n]` vs `\w`, etc... if we resolved these type issues we might have something workable here.

Reply:
> I was referring to \*, sorry for not being clearer. Someone went to the trouble of supporting it before making me reason perhaps it's more important than you think. :-) Obviously * inside bold doesn't require special treatment if it isn't triggering the end of the bold rule or causing any issues.


Let me check with Dan (who wrote the regular expression 8 years ago). Maybe at the time constrained formatting was not greedy... or maybe I'm missing something!

> The regex I gave you was working for simple cases, did you try it or play around with it in regex101? See the attached snap...

Not yet, I will give it a try tonight.

> So regarding broken lets go back to my prior example...
> This probably sort of works thanks to _ being in \w... So my apologies for not using a better example...
> Your PR would break all these that worked previously, would it not?

Let me try these examples on master but I'm pretty sure they do not work as expected today.
But rest assured, I plan to make them work but one step at a time :wink: 

> Go back and look at my comments regarding [^\n] vs \w, etc... if we resolved these type issues we might have something workable here.

:+1: